multiversx/mx-chain-go
# Feat/refactor nodes coordinator save config

## Reasoning behind the pull request
- Multiple nodes configs are being stored on each nodes coordinator registry key
  
## Proposed changes
- Refactor to store nodes configs per epoch on separate keys
- store nodes coordinator registry key as prefix + epoch instead of prefix + start of epoch header randomness
- move number of nodes config to store in config instead of static constant

## Testing procedure
- Standard system test
- we need to test if nodes are starting from local storage properly

Custom scenario:
- Set `NumEpochToKeep` from config.toml to a value larger than `NumNodesConfigEpochsToStore`:
       - example: NumEpochToKeep = 8 and NumNodesConfigEpochsToStore = 4
- Stop a node for more than 4 epochs (but less then 8) and see if it is able to recover

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
